Automattic/sensei
# Add/extensions section

Fixes #5630 

### Changes proposed in this Pull Request
* Reusing as maximum as possible from the previous implementation.
* Reorganised components structure to make more sense with the new design.
* Updated CSS to make it look like the design.
* ExtensionsProvider has been removed and `extensions` section from the Sensei Home REST API too.

### Testing instructions
* Checkout to the branch.
* Go to Sensei Home.
* Check that Extensions section look as the design.
* Check also responsive design.
* To test the update available scenario I suggest modifying the header for an installed plugin to an older version. In my case I changed `wp-content/plugins/sensei-certificates/woothemes-sensei-certificates.php` header to ` * Version: 1.3.0`.

### Screenshot / Video
#### Multiple extensions (multi-row) big screen.
<img width="1719" alt="image" src="https://user-images.githubusercontent.com/799065/195827596-0a07339e-11ca-40d0-8e42-47f26b575595.png">
#### Multiple extensions (multi-row) smaller screen.
<img width="1087" alt="image" src="https://user-images.githubusercontent.com/799065/195827670-2ab1008c-68db-4267-ad83-3dc7110476bf.png">
#### Multiple extensions smallest screen.
<img width="615" alt="image" src="https://user-images.githubusercontent.com/799065/195827741-b417247d-4edc-4f90-95d6-3109fa235f5b.png">
#### Update available
<img width="532" alt="image" src="https://user-images.githubusercontent.com/799065/195829570-feab368e-e545-45c1-a83d-7d907ab8148f.png">
